### ISSUE TITLE
More enhancements for Software center

### ISSUE THREAD

**Issue by madsrh** [ubuntu/yaru#465]
**EDIT** still to be done: 
- [x] installed list
- [ ] gray line at the top (in a coming PR)
- [ ] green installed-icon (not supported)
- [x] borders for buttons
- [ ] hover effect for categories and applications (slight background brightness increment, while box-shadow is blocked by g-s-c own gtk-style.css)
- [ ] change [text+icon color on press](https://github.com/ubuntu/gtk-communitheme/issues/465#issuecomment-396411194)

___

We've already done [some enhancements on Software Center](https://github.com/ubuntu/gtk-communitheme/issues/226), but I'm hoping you will agree that these small tweaks would be good too 😃 

1. **Installed** list:
Window background is #fafafa and list background is #ffffff. Could we add a discreet border to make the separation better? In the mockup below I've used the box-shadow ([like the reviews](https://github.com/ubuntu/gtk-communitheme/pull/230/files)).
![image](https://user-images.githubusercontent.com/3986894/40322689-7aa7016c-5d33-11e8-8764-2de8eec1e585.png)
 
---
2. As you also can see in the image above the gray scroll line is present even when I'm at the top of the list.
![image](https://user-images.githubusercontent.com/3986894/40322910-1ba71d40-5d34-11e8-88fb-7e1728e4ef48.png)

---
3. Since the [badges are green](https://github.com/ubuntu/gnome-shell-communitheme/issues/157), wouldn't it make sense to make installed apps green too?

![image](https://user-images.githubusercontent.com/3986894/40323003-6ad24a16-5d34-11e8-9e10-c7a2013191cd.png)

![image](https://user-images.githubusercontent.com/3986894/40784290-84196e42-64e5-11e8-95ea-3f9170468499.png)

---
4. Borders for buttons: Some _Cancel_ buttons already got this, but I would love to see it here as well
![image](https://user-images.githubusercontent.com/3986894/40323256-239476e6-5d35-11e8-90ac-feb17e0fa2e0.png)

**Comment by Feichtmeier** [comment]
This is so funny. You can style the .installed-icon in gtk inspector to color: green without a problem. When you try the same in _common.scss it just does not work. The only thing that works is to change the background to green, but this is not what we want, right?
I just love how much the gtk theming is working differently on different places! :dancing_men:

**Comment by Feichtmeier** [comment]
There is sadly no style class for the list :( If I style "list" every list is changed.
What would be possible is to style the 

 .list-box-app-row { border-radius: 5px; border: 1px solid $boders_color; margin: 10px 0;}
![screenshot from 2018-06-10 13-27-26](https://user-images.githubusercontent.com/15329494/41201116-4407504c-6cb2-11e8-9353-214d87706d60.png)
![screenshot from 2018-06-10 13-26-42](https://user-images.githubusercontent.com/15329494/41201117-44480510-6cb2-11e8-8ad0-43a583b0db20.png)

**Comment by Feichtmeier** [comment]
![image](https://user-images.githubusercontent.com/15329494/41205770-12cac1ce-6cfa-11e8-8ffa-d9eb64353e77.png)

**Comment by Feichtmeier** [comment]
@clobrano 
The border for the selection menu in software is very fat btw =) Maybe we can make a selection-menu PR?
![image](https://user-images.githubusercontent.com/15329494/41206004-a84534ac-6cfd-11e8-9b13-da7d005a6930.png)

**Comment by madsrh** [comment]
@Feichtmeier The "No screenshot provided" stays behind the image. Is this a bug?

I can remove it with `.screenshot-images { background: none; }` but this should only be applied when there's no preview image 🤷‍♂️ 

![image](https://user-images.githubusercontent.com/3986894/41229208-6cb91dc6-6d7b-11e8-80bc-01d7c8c115d1.png)

It does this in Adwaita too, but it looks like a mistake

**Comment by clobrano** [comment]
> I can remove it with .screenshot-images { background: none; } but this should only be applied when there's no preview image man_shrugging

this sounds like a bug

>  The border for the selection menu in photos is very fat btw =) Maybe we can make a selection-menu PR?

It should be equal to normal hb button, but it looks dark black, that's why it is weird
In selection-mode is without borders, right? We should make it the same (border or no border, but the same)

**Comment by madsrh** [comment]
> The "No screenshot provided" stays behind the image. Is this a bug?

Some applications looks really weird **without** the background - only a few (mostly the full height ones) looks better this way, so I guess the background should stay.

**Comment by madsrh** [comment]
Can we change buttons as well?

**For application buttons:** Adwaita has a very very highlight (of the text and background) on hover and currently we have nothing. But in the _Installed_ tab we use the hover gray.

Adwaita:
![peek 2018-06-11 15-17](https://user-images.githubusercontent.com/3986894/41260991-a3e921d4-6d8e-11e8-8565-4c7a78014b02.gif)

**Categories:** We have no hover and Adwaita again uses highlight (of icon, text and background). When pressed icon, text and border uses the selection color. I don't think that it looks very good with orange - especially for icon and text.


Community theme:
![peek 2018-06-11 15-47](https://user-images.githubusercontent.com/3986894/41261038-d5b66992-6d8e-11e8-8300-794091ebb705.gif)

**Comment by clobrano** [comment]
Little update over .installed-icon and the like. This style comes from gnome-software's gtk-style.css, which is a gresource. Looking for a way to overcome its priority over our css :+1:

**Comment by clobrano** [comment]
Currently we cannot change it. As said, it is defined in a css file which is loaded after ours and there's nothing we can do about it, except ask upstream if they can implement something to allow this, but for the future

**Comment by madsrh** [comment]
@clobrano Is there an upstream issue?

**Comment by clobrano** [comment]
Not yet, but technically it is not a bug, just a request

**Comment by Feichtmeier** [comment]
How about patching this downstream? Would this be possible? Basically disable this function call in the code and completely style it in our scsd

**Comment by clobrano** [comment]
which function call? We should patch Gnome Software

**Comment by Feichtmeier** [comment]
crazySoftwareCenterReturnData myCrazySoftwareCenterLoadCssFunction(AllmyFunnyArguments foobar) {
loadOrangeCssFiles("path");
loadGreenCssFiles("greenPath");
}

:laughing:

**Comment by clobrano** [comment]
@madsrh about your [comment](https://github.com/ubuntu/communitheme/issues/465#issuecomment-396411194) on **application** and **categories** buttons, I wonder if it was before the last button iteration.

We now have the same effect Adwaita has (background highlight), however, being our normal button color quite bright, it is less visible.
We should also have a increment of box-shadow on hover, but unfortunately box-shadow is set to **none** in the internal and unsuperable gtk-style.css :man_facepalming:

**Comment by madsrh** [comment]
@clobrano +1 sure, but 🤦‍♂️

**Comment by clobrano** [comment]
More a note to myself, I will try again to fix the gray line without introducing terrible side effects, but generally speaking I wouldn't spend more time on this than what already spent :smile:

**Comment by clobrano** [comment]
I am afraid, none of the requested changes are possible, due to custom css or too generic widgets to style :(

**Comment by madsrh** [comment]
Okay @clobrano 
Thanks for looking into this then 😩
